packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

```diff
@@ -5,47 +5,48 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, {
-  useState,
-  useEffect,
-  useContext,
-  useRef,
-  useCallback,
-} from 'react';
+import React, {useState, useRef, useCallback} from 'react';
 import classnames from 'classnames';
 
-import DocusaurusContext from '@docusaurus/context';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 
 import './styles.css';
 
+const loadJS = () => import('docsearch.js');
+let loadedJs = false;
+
 const Search = props => {
-  const [isEnabled, setIsEnabled] = useState(true);
+  const [initialized, setInitialized] = useState(false);
```

Comment:
Change to `useRef` instead. `initialized` is not used for rendering and doesn't need to be part of state.

Reply:
just use javascript boolean i guess :)